File: matlab/lib/+Ice/SlicedData.m
classdef SlicedData < handle
    %SLICEDDATA Holds class slices that cannot be unmarshaled because their types are not known locally.
    %
    %   SlicedData Properties:
    %     slices - The details of each slice, in order of most-derived to least-derived.

    % Copyright (c) ZeroC, Inc.

    properties(SetAccess=immutable)
        %SLICES The details of each slice, in order of most-derived to least-derived.
        %   cell array of SliceInfo objects
        slices (1, :) cell
    end
    methods(Hidden)
        function obj = SlicedData(slices)
            obj.slices = slices;
        end
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/SlicedData.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** Holds class slices that cannot be unmarshaled because their types are not known locally. */
public final class SlicedData {
    /**
     * Constructs a SlicedData from the given array of slices.
     *
     *  @param slices the slices of the unknown class, in order of most-derived to least-derived
     */
    public SlicedData(SliceInfo[] slices) {
        this.slices = slices;
    }

    /** The slices of the unknown class, in order of most-derived to least-derived. */
    public final SliceInfo[] slices;
}
